R/import.R

```diff
@@ -0,0 +1,144 @@
+    message("Processing prevalence")
+    prevalence <- readRDS(raw$prevalence)
+    ## The incoming raw data has *way* too much stuff in it; I've
+    ## asked Arran to remove it so that we get a lighter download.
+    prevalence <- prevalence[
+      prevalence$type == "prev" & prevalence$uncertainty == "mean",
+      !(names(prevalence) %in% c("type", "uncertainty"))]
+    i <- order(prevalence$index)
+    prevalence <- prevalence[i, ]
+    rownames(prevalence) <- NULL
+
+    tr <- c(netUse = "switch_nets",
+            irsUse = "switch_irs",
+            netType = "NET_TYPE")
+    prevalence <- rename(prevalence, unname(tr), names(tr))
+    prevalence$netType <- relevel(prevalence$netType, c(std = 1, pto = 2))
+    prevalence$intervention <- relevel(prevalence$intervention,
+                                       import_intervention_map())
+
+    saveRDS(prevalence, files$prevalence)
+  }
+
+  files
+}
+
+
+mintr_db_download <- function(path, overwrite = FALSE) {
+  info <- jsonlite::read_json(mintr_path("data.json"))
+
+  dest <- file.path(path, info$directory)
+  dir.create(dest, FALSE, TRUE)
+  for (f in info$files) {
+    if (overwrite || !file.exists(file.path(dest, f))) {
+      message(sprintf("Downloading '%s'", f))
+      download_file(file.path(info$root, info$directory, f),
+                    file.path(dest, f))
+    }
+  }
+  lapply(info$files, function(f) file.path(path, info$directory, f))
+}
+
+
+## Worked out with Arran by looking at their data
+import_translate_index <- function(index) {
+  remap <- list(
+    list(
+      from = "res1",
+      to = "levelOfResistance",
+      map = c("0%" = 0, "20%" = 20, "40%" = 40,
+              "60%" = 60, "80%" = 80, "100%" = 100)),
+    list(
+      from = "season",
+      to = "seasonalityOfTransmission",
+      map = c(seasonal = 1, perennial = 2)),
+    list(
+      from = "endem",
+      to = "currentPrevalence",
+      map = c(low = 0.05, med = 0.3, high = 0.6)),
+    list(
+      from = "phi",
+      to = "bitingIndoors",
+      map = c(high = 0.97, low = 0.78)),
+    list(
+      from = "Q0",
+      to = "bitingPeople",
+      map = c(high = 0.92, low = 0.74)),
+    list(
+      from = "nets",
+      to = "itnUsage",
+      map = c("0%" = 0.0, "20%" = 0.2, "40%" = 0.4,
+              "60%" = 0.6, "80%" = 0.8)),
+    list(
+      from = "sprays",
+      to = "sprayInput",
+      map = c("0%" = 0.0, "80%" = 0.8)))
+
+  for (x in remap) {
+    index <- rename(index, x$from, x$to)
+    index[[x$to]] <- relevel(index[[x$to]], x$map)
+  }
+
+  index
+}
+
+
+## Intervention mappings come from our metadata
+import_intervention_map <- function() {
+  config <- jsonlite::read_json(mintr_path("json/graph_prevalence_config.json"))
+  map <- character()
+  for (x in config$series) {
+    if (!is.null(x$id)) {
+      map[[x$id]] <- x$name
+    }
+  }
+  map
+}
+
+
+mintr_db_docker <- function(path) {
+  raw <- jsonlite::read_json(mintr_path("data.json"))$directory
+  mintr_db_open(path)
```

Comment:
Sorry finding this all a bit confusing - this call is just to (possibly) rebuild the database, we aren't actually using the returned db? But then we delete the `mintr.db` file so how does `main.R` then start the db? Doesn't it just rebuild it all over again on startup?

Perhaps I've totally misunderstood - I thought the intention was to bake the db into the image?

I'd be in favour of keeping `mintr_db_open` logically separate from `mintr_db_init`, so that it's clear when we're rebuilding/initialising, vs when we're opening an existing db.

Reply:
The database is ~1GB and inflates from the data fairly quickly, so this feels wasteful to bake into the image.